cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcProvisioner.java

```diff
@@ -122,13 +125,27 @@ public Cluster getClusterDetail(ProvisionerContext context,
   @Override
   public void initializeCluster(ProvisionerContext context, Cluster cluster) throws Exception {
     // Start the ZK server
-    try (SSHSession session = context.getSSHContext().createSSHSession(getMasterExternalIp(cluster))) {
+    try (SSHSession session = createSSHSession(context, getMasterExternalIp(cluster))) {
       LOG.debug("Starting zookeeper server.");
       String output = session.executeAndWait("sudo zookeeper-server start");
       LOG.debug("Zookeeper server started: {}", output);
     }
   }
 
+  private SSHSession createSSHSession(ProvisionerContext provisionerContext, String host) throws IOException {
+    try {
+      return provisionerContext.getSSHContext().createSSHSession(host);
+    } catch (IOException ioe) {
+      if (Throwables.getRootCause(ioe) instanceof ConnectException) {
+        throw new IOException(String.format(
+                "Failed to connect to host %s. Ensure that a GCP Firewall Rules exist that allow ssh and https" +
```

Comment:
We need only inbound rule for ssh and https right? Can we specify that if that's the case.